catalog/spatial/csw/spatial-csw-transformer/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/converter/AbstractGmdConverter.java

```diff
@@ -0,0 +1,150 @@
+
+    protected static final DatatypeFactory XSD_FACTORY;
+
+    protected static final TimeZone UTC_TIME_ZONE = TimeZone.getTimeZone("UTC");
+
+    private static final String XPATH_SEPARATOR = "/";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AbstractGmdConverter.class);
+
+    static {
+        DatatypeFactory factory = null;
+
+        try {
+            factory = DatatypeFactory.newInstance();
+        } catch (DatatypeConfigurationException e) {
+            LOGGER.debug("Failed to create xsdFactory", e);
+        }
+
+        XSD_FACTORY = factory;
+    }
+
+    protected AbstractGmdConverter() {
+
+        XStream xstream = new XStream(new Xpp3Driver(new NoNameCoder()));
+
+        xstream.setClassLoader(xstream.getClass()
+                .getClassLoader());
+
+        xstream.registerConverter(this);
+        getXstreamAliases().forEach(name -> xstream.alias(name, Metacard.class));
+    }
+
+    /**
+     * Get a list of aliases to be passed to {@link XStream#alias(String, Class)}.
+     *
+     * @return list of aliases
+     */
+    protected abstract List<String> getXstreamAliases();
+
+    @Override
+    public final boolean canConvert(Class clazz) {
+        return Metacard.class.isAssignableFrom(clazz);
+    }
+
+    @Override
+    public final Object unmarshal(HierarchicalStreamReader reader, UnmarshallingContext context) {
+        throw new NotImplementedException();
+    }
+
+    @Override
+    public final void marshal(Object source, HierarchicalStreamWriter inWriter,
+            MarshallingContext context) {
+        if (!(source instanceof Metacard)) {
+            LOGGER.debug("Failed to marshal Metacard: {}", source);
+            return;
+        }
+        MetacardImpl metacard = new MetacardImpl((Metacard) source);
+
+        PathTracker tracker = new PathTracker();
+        PathTrackingWriter trackingWriter = new PathTrackingWriter(inWriter, tracker);
+
+        XstreamPathValueTracker pathValueTracker = buildPaths(metacard);
+        XmlTree tree = buildTree(pathValueTracker.getPaths());
+
+        tree.accept(new XstreamTreeWriter(trackingWriter, tracker, pathValueTracker));
+
+    }
+
+    /**
+     * Builds up the xml paths and values to write.
+     * Order matters!  Paths should be added in the order they must be written.
+     *
+     * @param metacard must be non-null
+     * @return XstreamPathValueTracker containing XML paths and values to write
+     */
+    protected abstract XstreamPathValueTracker buildPaths(MetacardImpl metacard);
+
+    /**
+     * Get the name of the root name.
+     *
+     * @return root node
+     */
+    protected abstract String getRootNodeName();
+
+    protected final XmlTree buildTree(Set<Path> paths) {
+
+        XmlTree gmdTree = new XmlTree(getRootNodeName());
+
+        XmlTree current = gmdTree;
+
+        for (Path path : paths) {
+            String tree = path.toString();
+            XmlTree root = current;
```

Comment:
I'll think about it, it's not really my code, I don't want to risk mucking it up to just reduce one variable